### ISSUE TITLE
The LeetCode problem number is not the same as LeetHub-2.0

### ISSUE THREAD

**Issue by tzjsw** [raphaelheinz/LeetHub-3.0#3]
## The LeetCode problem number is not the same as LeetHub-2.0

**Describe the bug**
It seems that LeetCode problem number generated by LeetHub-3.0 is not the same as LeetHub-2.0. For example, when I use LeetHub-2.0 with old UI of LeetCode, I got the correct problem number for the following problems: 1004-max-consecutive-ones-iii and 1234-replace-the-substring-for-balanced-string. But when I use LeetHub-3.0 with new UI of LeetCode, I got the incorrect problem number: 1046-max-consecutive-ones-iii and 1351-replace-the-substring-for-balanced-string.

For problems with small numbers, it seems that LeetHub-3.0 works fine.

**Screenshots**
![Screen Shot 2023-12-03 at 1 33 05 PM](https://github.com/raphaelheinz/LeetHub-3.0/assets/28799243/666e243a-302e-4216-97ac-e6f70575bc8e)

**Comment by aystream** [comment]
No, I think you think LeetCode 2.0 worked, in fact there was the same error there https://github.com/arunbhardwaj/LeetHub-2.0/issues/41 .The id is taken from the id api, not the one that is there. But this is my assumption, I don't have time to look at it yet. 
<img width="977" alt="image" src="https://github.com/raphaelheinz/LeetHub-3.0/assets/4909904/3079dc8d-82e6-451b-b53e-0101e6ba06d8">

**Comment by raphaelheinz** [comment]
@aystream You are right. Thanks for your feedback. It is not an issue that was introduced with LeetHub-3.0. After checking my private leetcode-solution repo, I can confirm that this issue also occurred with Leetcode-2.0 using the new Leetcode UI.

The issue is due to Leetcode returns a wrong problem id. I found that the returned problem id from problem 1-741 is correct. From problem 743 onwards there are inconsistencies. 

Btw we can access the problem id in various ways. One possibility is to read the graphql query. Another way is to read the local Chrome cache value. Please take a look at the screenshot below. We can see that the id for problem 743 is stored incorrectly (stored as 744).

So this is a general leetcode problem over which we have no influence on client side. I think an idea to bypass this issue would be to parse the title and extract the id from there...But this is not a critical issue.

![shot](https://github.com/raphaelheinz/LeetHub-3.0/assets/28300143/9f85cfd2-83be-4d29-8ea2-4281a9610abf)

**Comment by hyunjinjeong** [comment]
@raphaelheinz 

Hi. This issue still occurs in the latest version (v0.0.13). Could you please check?
I'm using the new Leetcode UI.

Github repo
<img width="944" alt="image" src="https://github.com/user-attachments/assets/116892a2-9e2c-4c15-aa29-30c5e2ac37f1" />
<img width="400" alt="image" src="https://github.com/user-attachments/assets/0ed8c143-19de-49fe-b64b-61607079475c" />

Leethub version
<img width="693" alt="image" src="https://github.com/user-attachments/assets/c622e620-9403-4ea9-9292-95b23f3eca7b" />

Leetcode UI
<img width="1728" alt="image" src="https://github.com/user-attachments/assets/3d86fe1b-1926-4e5b-bd98-0d32216ae15c" />

**Comment by raphaelheinz** [comment]
@hyunjinjeong Do you know if there have been any recent changes in Leetcode UI that might be responsible this issue occurs again?

**Comment by hyunjinjeong** [comment]
@raphaelheinz 

AFIAK, there haven't been changes in the Leetcode UI, but I could be wrong.

**Comment by raphaelheinz** [comment]
I observed during the last days but could't reproduce yet. I assumed it's similar to our old issue caused by Leetcode's internal numbering change from some id on and higher. I will continue observing. Let me know if someone got more insights

@hyunjinjeong Is it correct you submitted Leetcode normally but two commits with different ids have been created?

**Comment by hyunjinjeong** [comment]
@raphaelheinz 

I usually click the manual push button after submitting the answer to make sure it is committed. So it might be related..?

But recently I haven't experienced the issue too. It seems like it only happens rarely. I'll let you know if it happens again!

**Comment by raphaelheinz** [comment]
I reproduced the issue lately on "1275. Find Winner on a Tic Tac Toe Game" which is internally handled as `questionId=1400` and `quesionFrontendId=1275`. Please see attached screenshot showing Leetcode graphql response.

In my repo both entries are existing. Obviously our parsing mechanism doesn't work reliable yet, sometimes it uses `questionId` instead of `quesionFrontendId`. We need to investigate more

<img width="778" alt="Image" src="https://github.com/user-attachments/assets/862fbba2-f338-44a4-b7f3-3e2c8393a163" />

**Comment by raphaelheinz** [comment]
Good news:) I have further investigated and found the issue. If the id cannot extracted from tab "description", a default value is used. So far we used "questiond" instead of "questionFrontend". I fixed in de7823a.

Please let me know in case you still reproduce in the future

**Comment by raphaelheinz** [comment]
I'm closing this as issue should be solved

**Comment by Cycle001** [comment]
@raphaelheinz 
There still have something wrong.

As following screenshot, not only the number but also the description is incorrect.

<img width="1593" height="588" alt="Image" src="https://github.com/user-attachments/assets/8a0e0c43-8d96-44ed-b497-acfec44bdc37" />

<img width="1107" height="617" alt="Image" src="https://github.com/user-attachments/assets/cb7fb319-60a6-4556-94b4-790033eab784" />

**Comment by Cycle001** [comment]
@raphaelheinz 

With the help of AI, I have fixed the issue of failing to retrieve problem descriptions and the incorrect problem numbers on the LeetCode Chinese website, as detailed in #91 and #92. I've submitted two pull requests and hope the author can review it.
